internal/manager/run.go

```diff
@@ -466,3 +466,31 @@ func readyzHandler(mgr manager.Manager, dataplaneSynchronizer IsReady) func(*htt
 		return nil
 	}
 }
+
+// GatewayClientsProvider is an interface that provides clients for the currently discovered Gateway instances.
+type GatewayClientsProvider interface {
+	GatewayClients() []*adminapi.Client
+}
+
+// SchemaServiceGetter returns schema service of an admin API client if there is any client available.
+type SchemaServiceGetter struct {
+	clientsManager GatewayClientsProvider
+}
+
+// NewSchemaServiceGetter creates a schema service getter that uses given client manager to maintain admin API clients.
+func NewSchemaServiceGetter(cm GatewayClientsProvider) SchemaServiceGetter {
+	return SchemaServiceGetter{
+		clientsManager: cm,
+	}
+}
+
+// GetSchemaService returns schema service of  an admin API client.
+// If there is no clients available, returns a fake schema service that always returns an error.
+func (ssg SchemaServiceGetter) GetSchemaService() kong.AbstractSchemaService {
+	clients := ssg.clientsManager.GatewayClients()
+	if len(clients) > 0 {
+		return clients[0].AdminAPIClient().Schemas
+	}
+	// returns a fake schema service when no gateway clients available.
+	return translator.UnavailableSchemaService{}
+}
```

Comment:
Nit: that could be placed in a separate file just for clarity

Reply:
moved to `schema_service.go`.